src/kabasuji/controllers/AddPieceMove.java

```diff
@@ -0,0 +1,47 @@
+package kabasuji.controllers;
+import kabasuji.models.Level;
+import kabasuji.models.PlacedPiece;
+
+/**
+ * Move action to add a piece to a level
+ * For use in the builder
+ * 
+ * @author Logan
+ *
+ */
+public class AddPieceMove implements IMove {
+  
+  
+  Level level;
+  
+  PlacedPiece piece;
+  
+  /**
+   * Constructor
+   * @param level the level that is being played
+   * @param piece the piece to add
+   */
+  public AddPieceMove(Level level, PlacedPiece piece) {
```

Comment:
Should this take a placed piece? or make a placed piece out of a piece?
I think the second one maybe